frontend/src/app/pages/exercises/exercise/shared/trainer-map-editor/trainer-map-editor.component.ts

```diff
@@ -75,4 +86,15 @@ export class TrainerMapEditorComponent {
     public setCurrentCategory(category: keyof typeof colorCodeMap) {
         this.currentCategory = category;
     }
+    public async vehicleOnMouseDown(
+        event: MouseEvent,
+        vehicleTemplate: VehicleTemplate
+    ) {
+        this.dragElementService.onMouseDown(event, {
+            type: 'vehicle',
+            template: vehicleTemplate,
+            materialTemplates: await firstValueFrom(this.materialTemplates$),
+            personnelTemplates: await firstValueFrom(this.personnelTemplates$),
```

Comment:
if you do it like this, you should get these synchronously via `getStateSnapshot`